functions.php

```diff
@@ -112,7 +115,76 @@ function get_instagram_feed($instagram_handle) {
     }
 
   }
-  //delete_transient( 'instagram_feed');
+  return $feed;
+}
+
+// Twitter Feed
+function get_twitter_feed($twitter_handle) {
+  
+  if( empty($twitter_handle) ) {
+    return new WP_ERROR('no-twitter-handle', 'Missing twitter handle');
+  }
+
+  $feed = get_transient( 'twitter_feed_' . $twitter_handle );
+
+  // If feed is not cached
+  if( empty($feed) ) {
+
+    // Require TwitterOAuth lib
+     if ( ! class_exists('TwitterOAuth')) {
+      require "lib/twitteroauth/autoload.php";
+     }
+
+    // Get keys
+    $twitter_key = IGV_get_option( '_igv_twitter_key' ); 
+    $twitter_secret = IGV_get_option( '_igv_twitter_secret' ); 
+
+    // Connect to twitter
+    $twitter = new TwitterOAuth($twitter_key, $twitter_secret);
+
+    // Get Timeline
+    $feed = $twitter->get('statuses/user_timeline', array(
+      'count'         =>  10,
+      'screen_name'   =>  $twitter_handle,
+      'trim_user'     => 'true',
+      'exclude_replies' => 'true',
+      'contributor_details' => 'false',
+      'include_rts'   => 'false'
+    ));
+
+    if( $feed->errors ) {
+      return false;
+    }
+
+    // Regex for URLs
+    $url_regex = "@(https?://([-\w\.]+[-\w])+(:\d+)?(/([\w/_\.#-]*(\?\S+)?[^\.\s])?)?)@";
+
+    foreach($feed as &$twit) {
+      // Remove URLs from the text
+      $twit->text = preg_replace($url_regex, '', $twit->text);
+
+      $urls = $twit->entities->urls;
+
+      $link = new StdClass();
+      
+      // If the twit links to a post inside the site, link that twit to that post
+      // else link it to twitter and set "blank" as TRUE
+      // we don't care about link to other sites inside the twit
+      if( strpos( $urls[0]->display_url,'8106') !== FALSE ) {
```

Comment:
surely this check needs to be 8106.tv otherwise a permalink that includes that numer [quite possible with a blog post about 8106 for example] will be caught by this